tests/test_config.py

```diff
@@ -0,0 +1,31 @@
+import os
+
+
+cwd = os.getcwd()
+if cwd not in {'root/project', '/opt/code/tap-square'}:
+    print("WARN: This script is meant to run from the top level directory of the tap.")
+
+print("Reading in filenames from tests directory.")
+print("Parsing directory for tests.")
+files = [
+    name for name in os.listdir(cwd + '/tests')
+    if 'test' == name[:4] and '.py' == name[-3:] and \
+    name not in ['test_client.py', 'test_config.py']
+]
+
+print("Reading contents of circle config")
+with open(cwd + "/.circleci/config.yml", "r") as config:
+    contents = config.read()
+
+# Check that each file in the directory can be found in a run black in the circle config
+print("Parsing circle config for run blocks.")
+runs = contents.replace(' ', '').replace('\n', '').split('-run:') # separate into run blocks
+matches = {f: False for f in files}
+for m in matches.keys():
+    print("Verifying {} is running in circle.".format(m))
+    if any([m in run for run in runs]):
+        matches[m] = True
```

Comment:
I think this for loop is a bit complicated. What value is it providing over just asserting all([file in contents for file in files])? Why split it into runs and create a dict of matches[file] to true/false?

Reply:
It's not a huge deal, the test is checking the right thing. I assume you tested it fails and succeeds correctly?

Reply:
Yeah I think this could use a refactor, it isn't super clear. I was a bit lazy and didn't want to spend too much time parsing the file, I also wasn't sure how to put in an assertion without failing too early. Like I didn't want to fail at the first missing file and skip the rest.

Reply:
Yes I tested both for circle and local runs.

Reply:
It's fine, better than not having a test there.